src/main/java/org/jabref/model/database/BibDatabase.java

```diff
@@ -193,21 +194,32 @@ public synchronized boolean insertEntry(BibEntry entry) throws KeyCollisionExcep
      * @return false if the insert was done without a duplicate warning
      */
     public synchronized boolean insertEntry(BibEntry entry, EntryEventSource eventSource) throws KeyCollisionException {
-        Objects.requireNonNull(entry);
+        insertEntries(Collections.singletonList(entry), eventSource);
+        return duplicationChecker.isDuplicateCiteKeyExisting(entry);
+    }
 
-        String id = entry.getId();
-        if (containsEntryWithId(id)) {
-            throw new KeyCollisionException("ID is already in use, please choose another");
-        }
+    public synchronized void insertEntries(List<BibEntry> entries) throws KeyCollisionException {
+        insertEntries(entries, EntryEventSource.LOCAL);
+    }
 
-        internalIDs.add(id);
-        entries.add(entry);
-        entry.registerListener(this);
+    private synchronized void insertEntries(List<BibEntry> newEntries, EntryEventSource eventSource) throws KeyCollisionException {
+        Objects.requireNonNull(newEntries);
 
-        eventBus.post(new EntryAddedEvent(entry, eventSource));
-        return duplicationChecker.isDuplicateCiteKeyExisting(entry);
+        for (BibEntry entry : newEntries) {
+            String id = entry.getId();
+            if (containsEntryWithId(id)) {
```

Comment:
This code is probably the reason why we are experienced performance issues when importing entries to larger DBs. Maybe we can find a better way for this check, but this does not have to be part of the current PR.